PennMobile/Home/HomeViewModel.swift

```diff
@@ -162,6 +162,41 @@ extension Optional {
             }
         }
         
+        async let wrappedTask: () = withCheckedContinuation { continuation in
+            OAuth2NetworkManager.instance.getAccessToken { token in
+                guard let token = token else {
+                    DispatchQueue.main.async {
+                        self.data.wrapped = .success(WrappedModel(semester: "", pages: []))
+                    }
+                    continuation.resume()
+                    return
+                }
+
+                let url = URLRequest(url: URL(string: "https://pennmobile.org/api/wrapped/semester/2025S/")!, accessToken: token)
+
+                let task = URLSession.shared.dataTask(with: url) { data, response, _ in
+                    guard let httpResponse = response as? HTTPURLResponse, let data = data, httpResponse.statusCode == 200 else {
```

Comment:
small thing that doesn't need to be blocked/changed, but you can just say `..., let data, ...` (as of [Swift 5.7](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0345-if-let-shorthand.md))